WordPress/Classes/ViewRelated/NUX/LoginFields.swift

```diff
@@ -0,0 +1,82 @@
+import Foundation
+
+/// LoginFields is a state container for user textfield input on the login screens
+/// as well as other meta data regarding the nature of a login attempt.
+///
+@objc
+class LoginFields: NSObject {
+    // These fields store user input from text fields.
+
+    /// Stores the user's account identifier (either email address or username) that is
+    /// entered in the login flow. By convention, even if the user is logging in
+    /// via an email address this field should store that value.
+    var username = ""
+
+    /// The user's password.
+    var password = ""
+
+    /// The site address if logging in via the self-hosted flow.
+    var siteAddress = ""
+
+    /// The two factor code entered by a user.
+    var multifactorCode = "" // 2fa code
+
+    /// Used by the SignupViewController. Signup currently asks for both a
+    /// username and an email address.  This can be factored away when we revamp
+    /// the signup flow.
+    var emailAddress = ""
+
+    var meta = LoginFieldsMeta()
```

Comment:
I had to think about `meta` as a property name. But, hmm, I think I like it 😉

Reply:
I originally had something like `metaFields` or `metaData` or something like that.  Seemed a bit silly when I stared at it for a bit so... yeah... meta. 🤷‍♂️